app/src/main/java/com/android/unio/ui/authentication/OverlayCard.kt

```diff
@@ -0,0 +1,113 @@
+package com.android.unio.ui.authentication
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Surface
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import com.android.unio.model.user.Interest
+import com.android.unio.ui.theme.AppTypography
+import kotlinx.coroutines.flow.MutableStateFlow
+
+@Composable
+fun InterestOverlay(
+    onDismiss: () -> Unit,
+    onSave: () -> Unit,
+    interests: MutableStateFlow<MutableList<Pair<Interest, MutableState<Boolean>>>>
```

Comment:
To reduce cognite complexity, parameter type could be changed to a `typealias`. This will ensure better readability.